doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,773 @@
+failure or possible reboot before the status is queried again.
+
+
+#### 3.4.1 Known check result inconsistencies vs. actual state
+There is a known but rarely and not naturally occurring issue where the health check will report an OK condition for a
+limited time but sending some messages might not be possible. This happens when an admin user logs out of the keys.
+
+The health check will detect if the tokens (the key containers) have not been signed into after `xroad-signer` startup.
+It will however, not detect immediately when the tokens are manually logged out of. The keys are cached by the `xroad-proxy`
+process for a short while. As long as the authentication key is still cached, the health check will return OK, even though
+the necessary signing context values for sending a message might no longer be cached. This means messages might fail to be sent
+even if the health check returns OK. As the authentication key expires from the cache (after a maximum of 5 minutes), the
+health check will start returning failures. This is a feature of caching and not a bug per se. In addition, logging out
+of a security server's keys should not occur by accident so it should not be a surprise that the node cannot send messages
+after not having access to it's keys.
+
+
+#### 3.4.2 Health check examples
+
+Before testing with an actual load balancer, you can test the health check service with `curl`, for example.
+
+Health check service response when everything is up and running and messages should go through.
+```
+$ curl -i localhost:5588
+   HTTP/1.1 200 OK
+   Content-Length: 0
+   Server: Jetty(8.y.z-SNAPSHOT)
+```
+
+Health check service response when the service `xroad-signer` is not running.
+```
+$ curl -i localhost:5588
+HTTP/1.1 500 Server Error
+Transfer-Encoding: chunked
+Server: Jetty(8.y.z-SNAPSHOT)
+
+Fetching health check response timed out for: Authentication key OCSP status
+```
+
+
+## 4. Database replication setup
+
+For technical details on the PostgreSQL replication, refer to the [official documentation](https://www.postgresql.org/docs/9.2/static/high-availability.html).
+Note that the versions of PostgreSQL distributed with RHEL and Ubuntu are different. At the time of writing, RHEL 7
+distributes PostgreSQL version 9.2 and Ubuntu 14.04 distributes version 9.3; the replication configuration is the same
+for both versions.
+
+
+### 4.1 Setting up TLS certificates for database authentication
+This section describes how to create and set up certificate authentication between the slave and master database instances.
+
+For further details on the certificate authentication, see the
+[PostgreSQL documentation](https://www.postgresql.org/docs/9.2/static/auth-methods.html#AUTH-CERT).
+
+
+1. Generate the Certificate Authority key and a self-signed certificate for the root-of-trust:
+
+   ```
+   openssl req -new -x509 -days 7300 -nodes -sha256 -out master.crt -keyout master.key -subj '/O=cluster/CN=master'
```

Comment:
Hmmm. Isn't "master" a very confusing name to use in relation to the CA keys? As we are configuring a master node and a slave node (nodes) here? It is easy to make the mistake of thinking that master.key is the key of the master node, which it is not (if I am still following...)